src/hmem_neuron.c

```diff
@@ -0,0 +1,192 @@
+/*
+ * (C) Copyright 2020 Hewlett Packard Enterprise Development LP
```

Comment:
nit: Did HPE really contribute to this file?

Reply:
No, but `hmem_neuron.c` is basically a copy/replace of `hmem_cuda.c`.